pulsar/internal/connection.go

```diff
@@ -428,7 +428,7 @@ func (c *connection) runPingCheck(pingCheckTicker *time.Ticker) {
 				// We have not received a response to the previous Ping request, the
 				// connection to broker is stale
 				c.log.Warn("Detected stale connection to broker")
-				c.Close()
+				c.cnx.Close()
```

Comment:
Why is this needed? The c.Close() closes underlying connection already. Same for the others below.